File: find_weird_chars.py
import os
import sys

def find_encoding_issues(directory):
    for root, d_names, f_names in os.walk(directory):
        if "node_modules" in d_names:
            d_names.remove("node_modules")
        if ".next" in d_names:
            d_names.remove(".next")
        for f in f_names:
            if not f.endswith((".ts", ".tsx", ".js", ".jsx", ".md")): continue
            path = os.path.join(root, f)
            try:
                with open(path, 'r', encoding='utf-8') as file:
                    content = file.read()
                    # Let's find exactly any non-ascii strings around words
                    for idx, line in enumerate(content.split('\n')):
                        has_strange = False
                        for char in line:
                            # Is non-ascii
                            if ord(char) > 127:
                                # Is not a standard spanish char or bullet or degree or dash or nbsp
                                if char not in 'áéíóúÁÉÍÓÚñÑüÜ¿¡•·°–— \u00A0':
                                    has_strange = True
                                    print(f"File: {path}, Line: {idx+1}")
                                    print(f"Char: {repr(char)}")
                                    print(f"Content: {line.strip()}")
                                    break
            except UnicodeDecodeError:
                print(f"Decode error on {path}")

if __name__ == "__main__":
    find_encoding_issues("c:/Users/CANVAS HALLANDALE 3/Downloads/Proyectos/cops-platform/cops-platform/apps/cotizaciones")

File: find_weird_chars.js
const fs = require('fs');
const path = require('path');

function walkDir(dir) {
    let results = [];
    const list = fs.readdirSync(dir);
    list.forEach(file => {
        file = path.join(dir, file);
        const stat = fs.statSync(file);
        if (stat && stat.isDirectory()) {
            if (!file.includes('node_modules') && !file.includes('.next')) {
                results = results.concat(walkDir(file));
            }
        } else {
            if (file.endsWith('.ts') || file.endsWith('.tsx') || file.endsWith('.json')) {
                results.push(file);
            }
        }
    });
    return results;
}

const dirPath = 'apps/cotizaciones';
const files = walkDir(dirPath);

files.forEach(file => {
    try {
        const content = fs.readFileSync(file, 'utf8');
        const lines = content.split('\n');

        lines.forEach((line, idx) => {
            // Find any character that is non-ASCII and NOT a standard Spanish accented character or common symbol
            const badMatch = line.match(/[^\x00-\x7FáéíóúÁÉÍÓÚñÑüÜ¿¡•·°–— \u00A0©]/g);
            if (badMatch) {
                console.log(`[WEIRD_CHAR] File: ${file}:${idx + 1} -> Chars: ${badMatch.join(' ')} -> Line: ${line.trim()}`);
            }
        });
    } catch (e) {
        console.error(`Could not read ${file}: ${e.message}`);
    }
});
